hnn_core/extracellular.py

```diff
@@ -517,13 +517,14 @@ def _build(self, cvode=None, include_celltypes='all'):
         # contributions of all segments on this rank to total calculated
         # potential at electrode (_PC.allreduce called in _simulate_dipole)
         self._nrn_voltages = h.Vector()
-
-        # Attach a callback for calculating the potentials at each time step.
-        # Enables fast calculation of transmembrane current (nA) at each
-        # segment. Note that this will run on each rank, so it is safe to use
-        # the extra_scatter_gather-method, which docs say doesn't support
-        # "multiple threads".
-        cvode.extra_scatter_gather(0, self._calc_potentials_callback)
+        # NB the vector is only initialised here: a separate callback-function
+        # must be attached to the _CVODE (extra_scatter_gather), which
+        # will append the summed potential of each electrode at each time step.
+        # The callback-function is included in this file (below), but must
+        # be called in network_builder during the actual simulation (including
+        # the callback as a method of the present class lead to segmentation
+        # violations when simulate_dipole was called multiple times in the same
```

Comment:
:grimacing: honestly this may have been my doing during the first pass at creating the LFP API.

Reply:
See the Issues I link to above. It's not at all clear why the initial implementation shouldn't work! And it _did_ work, just not repeatedly 🤷‍♂️